### ISSUE TITLE
[no-unused-vars] False positive related to `declare global`.

### ISSUE THREAD

**Issue by UNIDY2002** [typescript-eslint/typescript-eslint#2519]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
    "extends": [
        "plugin:@typescript-eslint/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "rules": {
        "@typescript-eslint/no-unused-vars": "error"
    }
}
```

```TS
export {};

declare global {
    interface Array<T> {
        toSet(): Set<T>;
    }
}

Array.prototype.toSet = function () {
    const set = new Set();
    this.forEach((value) => {
        set.add(value);
    });
    return set;
};

console.log([2, 3, 3].toSet());
```

```JSON
{
    "compilerOptions": {
        "module": "commonjs",
        "target": "es6"
    },
    "exclude": [
        "node_modules"
    ]
}
```

**Expected Result**

No lint error (as I think I was only mimicking the example at https://www.typescriptlang.org/docs/handbook/declaration-merging.html#global-augmentation).

**Actual Result**

```
  3:9  error  'global' is defined but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (1 error, 0 warnings)

```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.1.0` |
| `@typescript-eslint/parser`        | `4.1.0` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.1` |
| `node`                             | `12.13.1` |

**Comment by bradzacher** [comment]
I'm unable to repro this against master, or against a fresh 4.1.0 install.
Are you sure you're linting with the latest version of the plugin?

Could you please provide a repro repo to help investigate.

**Comment by UNIDY2002** [comment]
Great! Moving the declaration to a `.d.ts` file perfectly solved my problem!
I am actually quite new to TypeScript, and did not know much about these guidelines before. I will learn to use `.d.ts` files properly when programming in the future.

Thanks for your patience!

**Comment by mightyiam** [comment]
> Are you defining global augmentation within a normal `.ts` file?
> Whilst that's supported by TS, it's definitely not advised to do this.

I've been using TypeScript for a while and didn't learn or figure this out...

**Comment by bradzacher** [comment]
I think it's a mixed bag.
A lot of people define global module augmentations inline with their definitions.

If I've got time, I'll probably adjust the rule tomorrow to support this.
